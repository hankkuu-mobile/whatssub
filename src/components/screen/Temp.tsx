import {
  FlatList,
  Text,
} from 'react-native';
import React, { Component, useEffect } from 'react';
import { Button } from 'dooboo-native-widgets';
import { gql } from 'apollo-boost';
import styled from 'styled-components/native';
import { useQuery } from '@apollo/react-hooks';

export const QUERY_USERS = gql`
  query users {
    users {
      id
      name
      email
    }
  }
`;

const Container = styled.View`
  flex: 1;
  background-color: ${(props) => props.theme.background};
  flex-direction: row;
  align-items: center;
  justify-content: center;
`;

interface Props {
  navigation: any;
}

interface User {
  id: string;
  name: string;
  email: string;
}

const Users = () => {
  const { data, error, loading } = useQuery(QUERY_USERS);

  if (error) {
    return <Text>Error! {error.message}</Text>;
  }

  if (loading) {
    return <Text style={{
      color: 'rgba(255,255,255,07)',
      fontSize: 20,
    }}>Loading...</Text>;
  }

  return (
    <FlatList
      contentContainerStyle={{
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: 'blue',
      }}
      data={data.users ? data.users : []}
      keyExtractor={ (item: User) => String(item.id) }
      renderItem={({ item }) => <Text style={{
        color: 'rgba(255,255,255,07)',
        fontSize: 20,
      }}>{ item.email } - {item.name }</Text>}
    />
  );
};

function Page(props: Props) {
  return (
    <Container>
      <Users/>
    </Container>
  );
}

export default Page;
