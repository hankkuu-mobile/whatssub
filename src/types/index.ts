import { DefaultTheme } from 'styled-components';
/**
 * Fix error typing in `split` method in `apollo-link`
 * Related issue https://github.com/apollographql/apollo-client/issues/3090
 */
export interface Definition {
  kind: string;
  operation?: string;
};

export enum Gender { MALE, FEMALE };

export interface SocialInput {
  social: string;
  photo?: string;
  email?: string;
  password?: string;
  name?: string;
  nickname?: string;
  birthday?: Date;
  gender?: Gender;
  phone?: string;
}

export interface User {
  id: string;
  email: string;
  name: String;
  nickname: String;
  photo: String;
  birthday: Date;
  gender: Gender;
  social: String;
  verified: Boolean;
  notifications: [Notification]
  // customServices: [CustomService]
  // userProducts: [UserProduct]
  // reviews: [Review]
  created: Date
  updated: Date
  deleted: Date
}

export interface AuthPayload {
  token: string
  user: User
}

export enum ThemeType {
  LIGHT = 'LIGHT',
  DARK = 'DARK',
};

export interface ScreenProps {
  theme: DefaultTheme;
  changeTheme: () => void;
}
