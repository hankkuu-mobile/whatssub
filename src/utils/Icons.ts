export const IC_LOGO = require('../../assets/icons/logo.png');
export const IC_GOOGLE = require('../../assets/icons/google.png');
export const IC_FACEBOOK = require('../../assets/icons/facebook.png');
export const IC_SLASH = require('../../assets/icons/slash.png');
export const IC_NOTI_ON = require('../../assets/icons/alarmOn.png');
export const IC_NOTI_OFF = require('../../assets/icons/alarmOff.png');
export const IC_EDIT = require('../../assets/icons/edit.png');
export const IC_ADD = require('../../assets/icons/add.png');
export const IC_ADD_SERVICE = require('../../assets/icons/addService.png');
export const IC_WRITE = require('../../assets/icons/icWrite.png');
export const IC_LIST_BTN = require('../../assets/icons/icListBtn.png');
export const IC_BACK = require('../../assets/icons/back.png');
