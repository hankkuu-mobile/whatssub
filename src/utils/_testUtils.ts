import { ApolloClient } from 'apollo-client';
import { InMemoryCache } from 'apollo-cache-inmemory';
import { MockLink } from 'apollo-link-mock';

export function createClient(mocks) {
  return new ApolloClient({
    cache: new InMemoryCache(),
    link: new MockLink(mocks),
  });
}
