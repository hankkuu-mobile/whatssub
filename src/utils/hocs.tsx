import React from 'react';

function getDisplayName(WrappedComponent) {
  return WrappedComponent.displayName || WrappedComponent.name || 'Component';
}

export function withContainer<P>(injectedProps: P) {
  return function<T extends P>(WrappedComponent: React.FC<T>) {
    const hocComponent = (props: Omit<T, keyof P>) => (
      <WrappedComponent
        {...props as T} // this 'as T' should be explicit
        {...injectedProps}
      />
    );

    hocComponent.displayName = `WithContainer(${getDisplayName(
      WrappedComponent,
    )})`;

    return hocComponent;
  };
}
