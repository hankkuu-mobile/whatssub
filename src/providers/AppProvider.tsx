import * as Font from 'expo-font';

import React, { useEffect, useReducer, useState } from 'react';

import { AsyncStorage } from 'react-native';
import { ThemeType } from '../types';

interface Context {
  state: State;
  changeTheme: (theme?: ThemeType) => void;
}
const AppContext = React.createContext<Context>(null);
const AppConsumer = AppContext.Consumer;

interface ChangeThemeTypeAction {
  type: 'change-theme-type-mode';
  payload: {
    theme?: ThemeType;
  };
}

interface Props {
  theme?: ThemeType;
  children?: any;
  doNotWaitFont?: boolean;
}

interface State {
  theme?: ThemeType;
}

export const initialState: State = {
  theme: ThemeType.LIGHT,
};

const reducer = (state: State, action: ChangeThemeTypeAction) => {
  switch (action.type) {
    case 'change-theme-type-mode':
      // if payload.theme doesn't exist, toggle
      if (!action.payload.theme) {
        return {
          ...state,
          theme:
            state.theme === ThemeType.LIGHT
              ? ThemeType.DARK
              : ThemeType.LIGHT,
        };
      }
      // else change it with payload.theme
      return {
        ...state,
        theme: action.payload.theme,
      };
    default:
      return state;
  }
};

function AppProvider(props: Props) {
  const [state, dispatch] = useReducer(reducer, initialState);
  const [fontLoading, setFontLoading] = useState(true);

  // by encapsulating dispatch, it is easier to use (such as passing through screenProps)
  const changeTheme: (theme?: ThemeType) => void = (theme) => {
    dispatch({
      type: 'change-theme-type-mode',
      payload: {
        theme,
      },
    });
  };

  // initialize theme
  useEffect(() => {
    changeTheme(props.theme ? props.theme : initialState.theme);
  }, [props.theme]);

  // Load font and then use saved theme from local storage
  const getCurrentThemeType = async () => {
    const value = (await AsyncStorage.getItem('theme')) as ThemeType;
    value && changeTheme(value);
  };
  const loadFonts = async () => {
    await Font.loadAsync({
      'spoqa-han-sans-bold': require('../../assets/fonts/SpoqaHanSans/SpoqaHanSans-Bold.ttf'),
      'spoqa-han-sans-regular': require('../../assets/fonts/SpoqaHanSans/SpoqaHanSans-Regular.ttf'),
    });

    setFontLoading(false);
  };
  useEffect(() => {
    loadFonts();
    getCurrentThemeType();
  }, []);

  useEffect(() => {
    AsyncStorage.setItem('theme', state.theme);
  }, [state.theme]);

  return (
    <AppContext.Provider
      value={{
        state,
        changeTheme,
      }}
    >
      {fontLoading && !props.doNotWaitFont ? null : props.children}
    </AppContext.Provider>
  );
}

export { AppConsumer, AppProvider, AppContext };
