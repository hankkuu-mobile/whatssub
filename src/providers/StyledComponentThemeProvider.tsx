import React, { useContext } from 'react';

import { AppContext } from './AppProvider';
import PropTypes from 'prop-types';
import { ThemeProvider } from 'styled-components';
import { createTheme } from '../theme';

const StyledComponentThemeProvider = (props) => {
  const { state } = useContext(AppContext);
  return (
    <ThemeProvider theme={createTheme(state.theme)}>
      {props.children}
    </ThemeProvider>
  );
};

StyledComponentThemeProvider.propTypes = {
  children: PropTypes.node,
};

export default StyledComponentThemeProvider;
