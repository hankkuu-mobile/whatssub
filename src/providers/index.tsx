import { AppConsumer, AppContext, AppProvider } from './AppProvider';

import PropTypes from 'prop-types';
import React from 'react';
import StyledComponentThemeProvider from './StyledComponentThemeProvider';
import { ThemeType } from '../types';

interface Props {
  theme?: ThemeType;
  children?: any;
  doNotWaitFont?: boolean;
}

const InitialProviders = ({ children, ...rest }: Props) => (
  <AppProvider {...rest}>
    <StyledComponentThemeProvider>{children}</StyledComponentThemeProvider>
  </AppProvider>
);

InitialProviders.propTypes = {
  children: PropTypes.node,
};

export { AppProvider, AppConsumer, AppContext, InitialProviders };
