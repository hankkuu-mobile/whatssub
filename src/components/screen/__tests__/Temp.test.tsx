import 'react-native';
import * as React from 'react';
import {
  RenderResult,
  act,
  cleanup,
  fireEvent,
  render,
} from '@testing-library/react-native';
import Temp, {
  QUERY_USERS,
} from '../Temp';
import { ApolloProvider } from '@apollo/react-hooks';

import { AppProvider } from '../../../providers';
import { ThemeProvider } from 'styled-components/native';
import { ThemeType } from '../../../types';
import { createClient } from '../../../utils/_testUtils';
import { createTheme } from '../../../theme';
import renderer from 'react-test-renderer';

const createTestProps = (obj: object) => ({
  navigation: {
    navigate: jest.fn(),
    goBack: jest.fn(),
  },
  screenProps: {
    theme: jest.mock,
  },
  ...obj,
});

const mocks = [
  {
    request: {
      query: QUERY_USERS,
      // variables: {
      //   email: 'email@email.com',
      // },
    },
    result: {
      data: {
        users: [
          {
            id: 'user_id_1',
            email: 'user_email_1',
          },
          {
            id: 'user_id_2',
            email: 'user_email_2',
          },
        ],
      },
    },
  },
];

let props: any;
let component: React.ReactElement;
let testingLib: RenderResult;
let client;

afterEach(cleanup);

describe('[Temp] render', () => {
  beforeAll(() => {
    props = createTestProps({});
    client = createClient(mocks);
    component = (
      <AppProvider>
        <ThemeProvider theme={createTheme()}>
          <ApolloProvider client={client}>
            <Temp {...props} />
          </ApolloProvider>
        </ThemeProvider>
      </AppProvider>
    );
  });

  it('renders without crashing', () => {
    const rendered = renderer.create(component).toJSON();
    expect(rendered).toMatchSnapshot();
    expect(rendered).toBeTruthy();
  });
});

// describe('[Temp] Interaction', () => {
//   let renderResult: RenderResult;

//   beforeEach(() => {
//     renderResult = render(component);
//   });

//   it('should simulate [onClick] when [btn] has been clicked', () => {
//     const btnInstance = renderResult.getByTestId('btn');
//     fireEvent.press(btnInstance);
//     expect(props.navigation.goBack).toHaveBeenCalled();
//   });
// });
