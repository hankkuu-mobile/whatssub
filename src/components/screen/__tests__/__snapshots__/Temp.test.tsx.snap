// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Temp] render renders without crashing 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#f9fafb",
        "flexBasis": 0,
        "flexDirection": "row",
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <Text
    style={
      Object {
        "color": "rgba(255,255,255,07)",
        "fontSize": 20,
      }
    }
  >
    Loading...
  </Text>
</View>
`;
